src/main/java/core/basesyntax/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.BalanceOperationHandler;
+import core.basesyntax.operation.OperationHandler;
+import core.basesyntax.operation.PurchaseOperationHandler;
+import core.basesyntax.operation.ReturnOperationHandler;
+import core.basesyntax.operation.SupplyOperationHandler;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String SEPARATOR = ",";
+    private static final String FIRST_LINE_IN_OUTPUT = "fruit,quantity" + System.lineSeparator();
+    private OperationStrategy operationStrategy;
+
+    @Override
+    public String createFinalReport(Map<String, Integer> fruitsMap) {
+        StringBuilder builder = new StringBuilder(FIRST_LINE_IN_OUTPUT);
+        for (Map.Entry<String, Integer> map : fruitsMap.entrySet()) {
+            builder.append(map.getKey());
+            builder.append(SEPARATOR);
+            builder.append(map.getValue());
+            builder.append(System.lineSeparator());
+        }
+        return builder.toString();
+    }
+
+    @Override
+    public void editMapReportDependingOnOperation(List<FruitTransaction> fruits) {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyOperationHandler());
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        for (FruitTransaction fruit : fruits) {
+            operationStrategy.get(fruit.getOperation()).getNewValueDependingOnOperation(fruit);
+        }
```

Comment:
This part of code has nothing connected with name of method which it's placed in.
Please remember about single responsibility principle and make your method do only one thing it supposed to do